src/Repositories/BlockRepository.php

```diff
@@ -35,42 +33,36 @@ public function getCrops(string $role): array
 
     public function hydrate(TwillModelContract $model, array $fields): TwillModelContract
     {
-        if (Schema::hasTable(config('twill.related_table', 'twill_related'))) {
-            $relatedItems = Collection::make();
-
-            Collection::make($fields['browsers'])->each(function ($items, $browserName) use (&$relatedItems) {
-                Collection::make($items)->each(function ($item) use ($browserName, &$relatedItems) {
-                    try {
-                        // @todo: Repository could be null.
-                        $repository = $this->getModelRepository($item['endpointType'] ?? $browserName);
-                        $relatedItems->push(
-                            (object) [
-                                'related' => $repository->getById($item['id']),
-                                'browser_name' => $browserName,
-                            ]
-                        );
-                    } catch (ReflectionException $reflectionException) {
-                        Log::error($reflectionException);
-                    }
-                });
-            });
-
-            $model->setRelation('relatedItems', $relatedItems);
-        }
+        $relatedItems = collect($fields['browsers'])
+            ->flatMap(fn($items, $browserName) => collect($items)
+                ->map(fn($item, $position) => new RelatedItem([
+                    'subject_id' => $model->getKey(),
+                    'subject_type' => $model->getMorphClass(),
+                    'related_id' => $item['id'],
+                    'related_type' => $item['endpointType'],
+                    'browser_name' => $browserName,
+                    'position' => $position,
+                ])));
+
+        $model->setRelation('relatedItems', $relatedItems);
+        $model->loadMissing('relatedItems.related');
 
         return parent::hydrate($model, $fields);
     }
 
+    /** @param Block $model */
     public function afterSave(TwillModelContract $model, array $fields): void
     {
-        if (Schema::hasTable(config('twill.related_table', 'twill_related'))) {
+        if (!empty($fields['browsers'])) {
+            $browserNames = collect($fields['browsers'])->each(function ($items, $browserName) use ($model) {
+                // This will create items or delete them if they are missing
+                $model->saveRelated($items, $browserName);
+            })->keys();
+
+            // Delete all the related items that were emptied
+            RelatedItem::query()->whereMorphedTo('subject', $model)->whereNotIn('browser_name', $browserNames)->delete();
+        } else {
             $model->clearAllRelated();
```

Comment:
This would previously delete and recreate all related items, which is very write intensive on the database for nothing because saveRelated already does an existence check